src/Compilers/CSharp/Portable/Symbols/AliasSymbol.cs

```diff
@@ -334,41 +396,76 @@ private NamespaceOrTypeSymbol ResolveAliasTarget(NameSyntax? syntax, BindingDiag
             return declarationBinder.BindNamespaceOrTypeSymbol(syntax, diagnostics, basesBeingResolved).NamespaceOrTypeSymbol;
         }
 
-        public override bool Equals(Symbol? obj, TypeCompareKind compareKind)
+        internal override bool RequiresCompletion
         {
-            if (ReferenceEquals(this, obj))
+            get { return true; }
+        }
+    }
+
+    internal sealed class AliasSymbolFromTarget : AliasSymbol
```

Comment:
>AliasSymbolFromTarget [](start = 26, length = 21)

I'm not certain what `FromTarget` means here. Can you clarify?

Reply:
> I'm not certain what FromTarget means here. Can you clarify?

An alias has a "target", see AliasSymbol.Target property

---
In reply to: [583934465](https://github.com/dotnet/roslyn/pull/51505#discussion_r583934465) [](ancestors = 583934465)

Reply:
> An alias has a "target", see AliasSymbol.Target property

So, an alias of an alias? Is there any reason we can't just name this `AliasSymbolFromAlias`?

---
In reply to: [583936894](https://github.com/dotnet/roslyn/pull/51505#discussion_r583936894) [](ancestors = 583936894,583934465)

Reply:
From offline discussion, I'd prefer if this was renamed to something like `AliasSymbolFromResolvedTarget`, to more clearly convey the difference between these two types to readers.

---
In reply to: [583937547](https://github.com/dotnet/roslyn/pull/51505#discussion_r583937547) [](ancestors = 583937547,583936894,583934465)